packages/SymbioticController/src/Adapter/Nette/Routing/PresenterRoute.php

```diff
@@ -0,0 +1,124 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\SymbioticController\Adapter\Nette\Routing;
+
+use Nette\Application\IRouter;
+use Nette\Application\Request;
+use Nette\Http\IRequest;
+use Nette\Http\Url;
+use OdbavTo\PresenterRoute\RouteException;
+use Symplify\SymbioticController\Exception\MissingClassException;
+use Symplify\SymbioticController\Exception\MissingInvokeMethodException;
+
+final class PresenterRoute implements IRouter
+{
+    /**
+     * @var string
+     */
+    private $mask;
+
+    /**
+     * @var string
+     */
+    private $presenterClass;
+
+    public function __construct(string $mask, string $presenterClass)
+    {
+        $this->mask = $mask;
+        $this->setPresenterClass($presenterClass);
+    }
+
+    public function match(IRequest $httpRequest): ?Request
+    {
+        $urlScript = $httpRequest->getUrl();
+        $path = $this->normalizePath($urlScript->getPath());
+
+        $mask = $this->normalizePath($this->mask);
+        $mask = str_replace('/', '\/', $mask);
+
+        // use named sub-patterns to match params
+        $routeRegex = preg_replace('/<[\w_-]+>/', '(?$0[\w_-]+)', $mask);
+        $routeRegex = '@^' . $routeRegex . '$@';
+        $result = preg_match($routeRegex, $path, $matches);
```

Comment:
here we search for all "variables" (in form of <var>) and we make a named patterns from it - like (?<var>[\w_-]+) - so that we can use them right on path (with preg_match)